index.js

```diff
@@ -1184,6 +1184,34 @@ class Hyperdrive extends Nanoresource {
     })
   }
 
+  rename (oldpath, newpath, cb) {
+    this.lstat(oldpath, (err, oldstat) => {
+      if (err) return cb(err)
+      //TODO: Handle the following case : the new directory pathname contains a path prefix that names the old directory, should result in EINVAL
+      if (oldstat.isDirectory()) return cb(new errors.IsDirectory(oldpath)) //TODO: Support directory renaming
+      if (oldstat.linkName) return cb(new InvalidArgument(`${oldpath} not implemented yet`)) //TODO: Support symlink renaming
+      
+      this.exists(newpath, exists => {
+        if (exists) {
+          this.lstat(newpath, (err, newstat) => {
+            if (err) return cb(err)
+            if (!oldstat.isDirectory() && newstat.isDirectory()) return cb(new errors.IsDirectory(newpath))
+            if (oldstat.isDirectory() && !newstat.isDirectory()) return cb(new errors.IsNotDirectory(oldpath))
+            this.unlink(newpath, err => {
```

Comment:
That's what I believed, I since removed the unlink but now this [test](https://github.com/8000ff/hyperdrive/blob/95a2c923a5d19bd3aae218c6975939baa8b3f434/test/rename.js#L160) fails and shows that the content of the file was not copied.